### DISCUSSION TITLE
Can you explain the current and releases directory?

### DISCUSSION THREAD

**Discussion by rowanblub** [SjorsO/deploy-laravel#1] [Q&A]
Hi Sjors! 

I really like your deployment script, i am a beginner in deploying Laravel applications and CI/CD in general and this is a really good starting point for me.

However, i was wondering, could you explain the current and releases directories and where they are for? I had trouble finding it in the code.

I understand current is the most up-to-date production version of the app. But that is how far i've come.

Thanks!

**Comment by SjorsO** [comment]
I've just added an explanation [in the readme](https://github.com/SjorsO/deploy-laravel?tab=readme-ov-file#zero-downtime-deployments-current--releases-directory)

Let me know if there is anything else that isn't clear. I'd be glad to clarify.

**Reply by rowanblub** [reply]
Amazing! Thanks for the explanation. I came across another thing i'd like to know. In my local development environment, i am storing a decorative image (used for the login screen) in the public/storage directory, as multiple sources instructed me to do so. However this image does not get deployed this way.

What is the preferred way of storing images like this? And user-uploaded images for in the future?

**Reply by SjorsO** [reply]
You can store images and other assets in your public directory no problem. If you do it this way you should commit them to git as well.

The `public/storage` directory is actually the only directory where you can't store assets, because this directory is used by `Storage::disk('public')`. The deployment script calls `artisan storage:link` which creates a symlink from `storage/app/public` to `public/storage`. If you store your images in `public/images` for example it should work fine.

As for user uploaded files, always use the `Storage` facade when storing these. Any files you put in `/public` will get deleted when you deploy a new release.

**Reply by rowanblub** [reply]
Thanks for the explanation! You have a donation link anywhere? ;)

**Reply by SjorsO** [reply]
Don't worry about it 👍
